File: golang/mergeSortedArray.go
package main

func Merge(nums1 []int, m int, nums2 []int, n int) []int {
	i := n - 1
	j := n - 1
	k := m + n - 1
	for j >= 0 {
		num1 := nums1[i]
		num2 := nums2[j]
		if i >= 0 && num1 < num2 {
			nums1[k] = num2
			j--
		} else {
			nums1[k] = num1
			i--
		}
		k--
	}
	return nums1
}

File: typescript/mergeSortedArray.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
  let i = m - 1;       // last valid element in nums1
  let j = n - 1;       // last element in nums2
  let k = m + n - 1;   // last position in nums1

  while (j >= 0) {
    if (i >= 0 && nums1[i]! > nums2[j]!) {
      nums1[k] = nums1[i]!
      i--
    } else {
      nums1[k] = nums2[j]!
      j--
    }
    k--
  }
  console.log({nums1})
}

merge([1,2,3,0,0,0],3,[2,5,6],3)